server/src/main/java/org/opensearch/common/inject/internal/Errors.java

```diff
@@ -661,8 +683,7 @@ public static void formatInjectionPoint(Formatter formatter, Dependency<?> depen
 
         } else if (dependency != null) {
             formatter.format("  while locating %s%n", convert(dependency.getKey()));
-            formatter.format("    for parameter %s at %s%n",
-                    dependency.getParameterIndex(), StackTraceElements.forMember(member));
+            formatter.format("    for parameter %s at %s%n", dependency.getParameterIndex(), StackTraceElements.forMember(member));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getParameterIndex&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 686. Consider formatting this data with &#39;%d&#39; instead.